repowiz-infrastructure/src/main/groovy/life/qbic/repowiz/finalise/geo/GeoTemplateParser.groovy

```diff
@@ -0,0 +1,85 @@
+package life.qbic.repowiz.finalise.geo
+
+
+import life.qbic.repowiz.finalise.parsing.RepositoryInput
+import life.qbic.repowiz.finalise.parsing.RepositoryOutput
+import life.qbic.repowiz.io.XlsxParser
+import life.qbic.repowiz.prepare.model.RepoWizProject
+import life.qbic.repowiz.prepare.model.RepoWizSample
+import org.apache.poi.xssf.usermodel.XSSFCell
+
+class GeoTemplateParser extends XlsxParser implements RepositoryInput {
+
+    final byte[] rgbLevelColor = [-1, 0, 0]
+    final byte[] rgbFieldColor = [0, 0, -1]
+
+    RepositoryOutput output
+
+    GeoTemplateParser(){ //e.g hts ["METADATA TEMPLATE"], affymetrix_ge ["METADATA","MATRIX"] oder so
+        super(["METADATA TEMPLATE"])
+        super.commentMarker = '#'
+        super.mapper = new GeoMapper("hts")
```

Comment:
Maybe a static factory method is a more intuitive approach here? For example:

```
GeoMapper htsMapper = GeoMapper.createHTSMapper()
GeoMapper xxxMapper = GeoMapper.createXXXMapper()
```
What do you think?

Reply:
you are right, this is adjusted within the introduced module geo see feature/spi :)